crates/qt-build-utils/src/lib.rs

```diff
@@ -343,35 +417,35 @@ impl QtBuild {
             } else {
                 (
                     format!("Qt{}{qt_module}", self.version.major),
-                    format!(
-                        "{}/{}Qt{}{}.prl",
-                        lib_path, prefix, self.version.major, qt_module
-                    ),
+                    self.find_qt_module_prl(&lib_path, prefix, self.version.major, qt_module),
                 )
             };
 
-            println!("cargo:rustc-link-lib={link_lib}");
-
-            match std::fs::read_to_string(&prl_path) {
-                Ok(prl) => {
-                    for line in prl.lines() {
-                        if let Some(line) = line.strip_prefix("QMAKE_PRL_LIBS = ") {
-                            parse_cflags::parse_libs_cflags(
-                                &format!("Qt{}{qt_module}", self.version.major),
-                                line.replace(r"$$[QT_INSTALL_LIBS]", &lib_path)
-                                    .replace(r"$$[QT_INSTALL_PREFIX]", &lib_path)
-                                    .as_bytes(),
-                            );
-                        }
-                    }
-                }
-                Err(e) => {
-                    println!(
-                        "cargo:warning=Could not open {} file to read libraries to link: {}",
-                        &prl_path, e
-                    );
-                }
-            }
+            self.cargo_link_qt_library(
+                builder,
+                &format!("Qt{}{qt_module}", self.version.major),
+                &prefix_path,
+                &lib_path,
+                &link_lib,
+                &prl_path,
+            );
+        }
+
+        let emscripten_targeted = match env::var("CARGO_CFG_TARGET_OS") {
+            Ok(val) => val == "emscripten",
+            Err(_) => false,
+        };
+        if emscripten_targeted {
```

Comment:
Can't this be condensed to a single `if let` statement?

Reply:
I could condense it down to a single if let statement but it doesn't look really nice to me
`if let Ok(true) = env::var("CARGO_CFG_TARGET_OS").map(|val| val == "emscripten") {`

I tired doing it with 2 if statements but this changes the indenting and adds some linebreaks after `cargo fmt`
```
        if let Ok(target_os) = env::var("CARGO_CFG_TARGET_OS").map(|val| val == "emscripten") {
            if target_os == "emscripten" {
                let platforms_path =
                    format!("{}/platforms", self.qmake_query("QT_INSTALL_PLUGINS"));
                println!("cargo:rustc-link-search={platforms_path}");
                self.cargo_link_qt_library(
                    "qwasm",
                    &prefix_path,
                    &lib_path,
                    "qwasm",
                    &format!("{platforms_path}/libqwasm.prl"),
                    builder,
                );
            }
        }
```

Reply:
Okay, I'm not going to nitpick on this.